tenable/ot/graphql/definitions.py

```diff
@@ -63,77 +65,107 @@ def __str__(self):
+        return self.snake_case_regex2.sub(r"\1_\2", name).lower()
 
     @post_load
-    def to_object(self, data, **kwargs):
-        '''This method turns the schema into its corresponding object.'''
-        return Extensions(**data)
+    def to_object(
+        self,
+        data: dict,
+        *args,
+        **kwargs
+    ):
+        """This method turns the schema into its corresponding object."""
+        if self.dataclass is None:
+            raise ValueError(f"dataclass must be set for {type(self)}")
+        if not isinstance(data, dict):
+            raise ValueError("data must be of type dict")
+        normalized_data = {
+            self.camel_to_snake(key): value for key, value in data.items()
+        }
 
+        return self.dataclass(**normalized_data)
 
-class GraphqlErrorSchema(Schema):
-    '''
-    Schema for GraphQL error.
-    '''
-    message = fields.Str(required=True)
-    locations = fields.List(fields.Nested(LocationSchema))
-    path = fields.List(fields.Str())
-    extensions = fields.Nested(ExtensionsSchema)
 
-    @post_load
-    def to_object(self, data, **kwargs):
-        '''This method turns the schema into its corresponding object.'''
-        return GraphqlError(**data)
+class NodesSchema(SchemaBase):
+    def __iter__(self):
+        yield from self.nodes
 
 
 @dataclass
-class Location():
-    '''
+class Location:
+    """
     This class holds the location part of the error.
-    '''
+    """
+
     line: int
     column: int
 
 
 @dataclass
-class Extensions():
-    '''
+class Extension:
+    """
     This class holds the extensions part of the error.
-    '''
+    """
+
     code: str = None
 
 
 @dataclass
 class GraphqlError(Exception):
-    '''
+    """
     This class holds a GraphQL error.
-    '''
+    """
+
     message: str
     locations: List[Location] = None
     path: List[str] = None
-    extensions: Extensions = None
+    extensions: Extension = None
 
     def __str__(self):
-        return '{ret_code}: {message} ({location}), path {path}'.format(
-            ret_code=self.extensions.code,
-            message=self.message,
-            location=self.locations[0],
-            path=self.path
-        )
+        return f"{self.extensions.code}: {self.message} ({self.locations[0]}), path {self.path}"
+
+
+class LocationSchema(SchemaBase):
+    """
+    Schema for GraphQL location part of the error.
+    """
+
+    dataclass = Location
+
+    line = fields.Int(required=True)
+    column = fields.Int(required=True)
+
+
+class ExtensionsSchema(SchemaBase):
+    """
+    Schema for GraphQL extensions part of the error.
+    """
+
+    dataclass = Extension
+    code = fields.Str()
+
+
+class GraphqlErrorSchema(Schema):
+    """
+    Schema for GraphQL error.
+    """
+
+    message = fields.Str(required=True)
+    locations = fields.List(fields.Nested(LocationSchema))
+    path = fields.List(fields.Str())
+    extensions = fields.Nested(ExtensionsSchema)
```

Comment:
shouldn't locations and extensions be set to allow none as they are in an exception?